include/npcomp/Dialect/Torch/IR/TorchOps.td

```diff
@@ -366,6 +367,34 @@ def Torch_PrimListConstructOp: Torch_Op<"prim.ListConstruct", [
   }];
 }
 
+def Torch_PrimDictConstructOp: Torch_Op<"prim.DictConstruct", [
+    NoSideEffect,
+    AllowsTypeRefinement,
+    SameVariadicOperandSize,
+    ]> {
+  let summary = "TorchScript prim::DictConstruct op";
+
+  let arguments = (ins
+    Variadic<AnyTorchDictKeyType>:$keys,
+    Variadic<AnyTorchType>:$values
+  );
+
+  let results = (outs
+    Torch_DictType:$result
+  );
+
+  let verifier = "return ::verify(*this);";
+
+  let assemblyFormat = [{
+    (`(``keys` $keys^ `values` $values`)`)? attr-dict `:` functional-type(operands, results)
```

Comment:
I think it would be a bit nicer if the format was

```
`keys` `(` ($keys^ `:` type($keys))? `)` `values` `(` ($values^ `:` type($values))? `)` attr-dict `:` type($results)
```